modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts

```diff
@@ -0,0 +1,230 @@
+			{
+				specificationKey,
+
+				value: {
+					en_US: productName,
+				},
+			},
+		],
+		productStatus: 0,
+		productType: 'simple',
+	};
+
+	return postProduct;
+};
+
+const getSpecification = (
+	optionCategoryId: number,
+	optionCategoryName: number,
+	priority: number
+) => {
+	const postSpecification = {
+		facetable: true,
+		key: optionCategoryName,
+		optionCategory: {
+			id: optionCategoryId,
+			key: optionCategoryName,
+			priority,
+			title: {
+				en_US: optionCategoryName,
+			},
+		},
+		title: {
+			en_US: optionCategoryName,
+		},
+	};
+
+	return postSpecification;
+};
+
+test('can sort specifications by specification group priority', async ({
+	_apiHelpers,
+	_layoutsPage,
+	_specificationFacetsPage,
+}) => {
+	const pageLabel = 'Specification Facet Page';
+
+	await _layoutsPage.createWidgetPage(pageLabel);
+
+	await expect(
+		_layoutsPage.page
+			.getByTestId('headerTitle')
+			.filter({hasText: pageLabel})
+	).toBeVisible({timeout: 60 * 1000});
+
+	await _specificationFacetsPage.goToPage();
+	await _specificationFacetsPage.addSearchOptionsWidget();
+	await _specificationFacetsPage.addSpecificationFacetWidget();
+
+	const site = await _apiHelpers.headlessAdminUser.getSiteByFriendlyUrlPath(
+		'guest'
+	);
+
+	const channel = await _apiHelpers.headlessCommerceAdminChannel.postChannel(
+		getChannel('Specification Facet Channel', site.id)
+	);
+
+	const optionCategory1 =
+		await _apiHelpers.headlessCommerceAdminCatalog.postOptionCategory(
+			getOptionCategory('Warranty', 0)
+		);
+	const optionCategory2 =
+		await _apiHelpers.headlessCommerceAdminCatalog.postOptionCategory(
+			getOptionCategory('Material', 1)
+		);
+
+	const specification1 =
+		await _apiHelpers.headlessCommerceAdminCatalog.postSpecification(
+			getSpecification(
+				optionCategory1.id,
+				optionCategory1.key,
+				optionCategory1.priority
+			)
+		);
+	const specification2 =
+		await _apiHelpers.headlessCommerceAdminCatalog.postSpecification(
+			getSpecification(
+				optionCategory2.id,
+				optionCategory2.key,
+				optionCategory2.priority
+			)
+		);
+
+	const catalog = await _apiHelpers.headlessCommerceAdminCatalog.postCatalog(
+		getCatalog('Specification Facet Catalog')
+	);
+
+	const product1 = await _apiHelpers.headlessCommerceAdminCatalog.postProduct(
+		getProduct(catalog.id, 'Product1', specification1.key)
+	);
+	const product2 = await _apiHelpers.headlessCommerceAdminCatalog.postProduct(
+		getProduct(catalog.id, 'Product2', specification2.key)
+	);
+
+	await _specificationFacetsPage.configureSearchOptions();
+
+	await _specificationFacetsPage.goToPage();
+
+	await expect(
+		_layoutsPage.page
+			.getByTestId('headerTitle')
+			.filter({hasText: pageLabel})
+	).toBeVisible({timeout: 60 * 1000});
+
+	await expect(
+		_layoutsPage.page.getByTestId(optionCategory1.key + '1')
+	).toBeVisible({timeout: 60 * 1000});
+
+	await expect(
+		_layoutsPage.page.getByTestId(optionCategory2.key + '2')
+	).toBeVisible({timeout: 60 * 1000});
```

Comment:
I still don't understand how the sorting is guaranteed here? I was expecting something like:
- find the list of specification by data-qa-id
- loop the entries of the the list
- verify each element to be in the right position

Reply:
Done as requested. 

```
	const panelList = await _specificationFacetsPage.panelList.all();

	const optionCategoryList = ['material', 'warranty'];

	for (let i = 0; i < optionCategoryList.length; i++) {
		await expect(panelList[i]).toHaveText(optionCategoryList[i]);
	}
```